cypress/integration/circulation-log/search.spec.js

```diff
@@ -2,62 +2,83 @@ import TopMenu from '../../support/fragments/topMenu';
-              materialType: { id: Cypress.env('materialTypes')[0].id },
-            }],
-          ],
-        });
-      })
-      .then(() => {
-        cy.createItemCheckout({
-          itemBarcode: ITEM_BARCODE,
-          userBarcode: Cypress.env('users')[0].barcode,
-          servicePointId: Cypress.env('userServicePoints')[0].id,
-        });
+    cy.createTempUser([
+      permissions.inventoryAll.gui,
+      permissions.circulationLogAll.gui
+    ])
+      .then(userProperties => {
+        userId = userProperties.userId;
+        cy.login(userProperties.username, userProperties.password);
+        cy.visit(TopMenu.circulationLogPath);
+        cy.getToken(Cypress.env('diku_login'), Cypress.env('diku_password'))
+          .then(() => {
+            cy.getLoanTypes({ limit: 1 });
+            cy.getMaterialTypes({ limit: 1 });
+            cy.getLocations({ limit: 1 });
+            cy.getHoldingTypes({ limit: 1 });
+            cy.getHoldingSources({ limit: 1 });
+            cy.getInstanceTypes({ limit: 1 });
+            cy.getServicePointsApi({ limit: 1, query: 'pickupLocation=="true"' });
+            cy.getUsers({
+              limit: 1,
+              query: `"personal.lastName"="${userProperties.username}" and "active"="true"`
+            });
+          })
+          .then(() => {
+            cy.addServicePointToUser(Cypress.env('servicePoints')[0].id, userId);
+            cy.getUserServicePoints(Cypress.env('users')[0].id);
+            cy.createInstance({
+              instance: {
+                instanceTypeId: Cypress.env('instanceTypes')[0].id,
+                title: `Checkout instance ${Number(new Date())}`,
+              },
+              holdings: [{
+                holdingsTypeId: Cypress.env('holdingsTypes')[0].id,
+                permanentLocationId: Cypress.env('locations')[0].id,
+                sourceId: Cypress.env('holdingSources')[0].id,
+              }],
+              items: [
+                [{
+                  barcode: ITEM_BARCODE,
+                  missingPieces: '3',
+                  numberOfMissingPieces: '3',
+                  status: { name: 'Available' },
+                  permanentLoanType: { id: Cypress.env('loanTypes')[0].id },
+                  materialType: { id: Cypress.env('materialTypes')[0].id },
+                }],
+              ],
+            });
+          })
+          .then(() => {
+            cy.createItemCheckout({
+              itemBarcode: ITEM_BARCODE,
+              userBarcode: Cypress.env('users')[0].barcode,
+              servicePointId: Cypress.env('userServicePoints')[0].id,
+            });
+          });
       });
   });
 
   afterEach('reset search results', () => {
     SearchPane.resetResults();
   });
 
+  after('check-in and delete user', () => {
+    cy.createItemCheckinApi({
+      itemBarcode: ITEM_BARCODE,
+      servicePointId: Cypress.env('servicePoints')[0].id,
+      checkInDate: '2021-09-30T16:14:50.444Z',
```

Comment:
can we use current date minus required days count?

Reply:
Why? In this I create record with any date then delete it after 30 seconds :)